commit 710765a87d599d95de51b79202ba3d82fd03ed95
Author: Steve Dickson <steved@redhat.com>
Date:   Wed Jun 25 09:23:45 2008 -0400

    When a FQDN exists in /var/lib/nfs/rmtab it causes
    the exportfs command to seg fault due to the nfs_export pointer
    not being allocated. Reworking the parentheses in rmtab_read()
    so the htype variable is evaluated correctly fix the problem.
    
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/support/export/rmtab.c b/support/export/rmtab.c
index 0ce3682..e11a22a 100644
--- a/support/export/rmtab.c
+++ b/support/export/rmtab.c
@@ -31,10 +31,10 @@ rmtab_read(void)
 		int			htype;
 		
 		htype = client_gettype(rep->r_client);
-		if (htype == MCL_FQDN || (htype == MCL_SUBNETWORK
+		if ((htype == MCL_FQDN || htype == MCL_SUBNETWORK)
 		    && (hp = gethostbyname (rep->r_client))
 		    && (hp = hostent_dup (hp),
-			(exp = export_allowed (hp, rep->r_path))))) {
+			exp = export_allowed (hp, rep->r_path))) {
 			/* see if the entry already exists, otherwise this was an instantiated
 			 * wild card, and we must add it
 			 */
